src/main/java/org/generationcp/middleware/dao/GermplasmSearchDAO.java

```diff
@@ -691,17 +692,41 @@ protected String getValueOfAddedColumns(final String propertyId, final Object[]
 			}
 		}
 
-		return this.getTypesFromUserDefinedFieldTable(addedColumnsPropertyIds, nonStandardColumns, "NAMES");
+		return this.getNameTypesFromUserDefinedFieldTable(addedColumnsPropertyIds, nonStandardColumns);
 	}
 
-	private Map<String, Integer> getTypesFromUserDefinedFieldTable(final List<String> addedColumnsPropertyIds,
-		final List<String> nonStandardColumns, final String ftable) {
+	private Map<String, Integer> getNameTypesFromUserDefinedFieldTable(final List<String> addedColumnsPropertyIds,
+		final List<String> nonStandardColumns) {
 		final Map<String, Integer> typesMap = new HashMap<>();
-		final String field = GermplasmSearchDAO.ATRIBUTS.equals(ftable) ? "fcode" : "fname";
 		if (!nonStandardColumns.isEmpty()) {
 			final SQLQuery query = this.getSession()
-				.createSQLQuery("SELECT " + field + ", fldno from udflds where ftable = :ftable and " + field + " IN (:fieldList)");
-			query.setParameter("ftable", ftable);
+				.createSQLQuery("SELECT fname , fldno from udflds where ftable = 'NAMES' and fname IN (:fieldList)");
+			query.setParameterList("fieldList", addedColumnsPropertyIds);
+			final List<Object[]> results = query.list();
+
+			for (final Object[] row : results) {
+				typesMap.put(String.valueOf(row[0]).toUpperCase(), (Integer) row[1]);
+			}
+
+		}
+
+		return typesMap;
+	}
+
+	private Map<String, Integer> getAttributeFromOntology(final List<String> addedColumnsPropertyIds,
+		final List<String> nonStandardColumns, final String programUUID) {
+		final Map<String, Integer> typesMap = new HashMap<>();
+		if (!nonStandardColumns.isEmpty()) {
+			final SQLQuery query = this.getSession()
+				.createSQLQuery("SELECT if(vo.alias is null,cv.name ,vo.alias) AS name, cv.cvterm_id "
+					+ " FROM cvterm cv INNER JOIN cvtermprop cp ON cp.type_id = " + TermId.VARIABLE_TYPE.getId() + " and cv.cvterm_id = cp.cvterm_id " //
+					+ " LEFT JOIN variable_overrides vo ON vo.cvterm_id = cv.cvterm_id AND vo.program_uuid = :programUUID " //
```

Comment:
Move the filtering of variable types to main clause instead of where clause:
`INNER JOIN cvterm vartype on vartype.cvterm.name = cp.value and vartype.cvterm_id in (...)`

Reply:
done.